src/App.tsx

```diff
@@ -1,14 +1,34 @@
 /* eslint-disable max-len */
-import React from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { useEffect, useState } from 'react';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions } from './features/todos';
 
 export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const [loading, setLoading] = useState<boolean>(true);
```

Comment:
Not sure that you actually need this loading state here. 
Working with redux, most probably, you will have something like todosLoading in your store that will be toggled through actions. 

However, to emulate this approach here, I would also add a `setLoading(true)` in `useEffect` before fetching. 
This will be more explicit approach, without having to consider a default state.